xbmc/pvr/dialogs/GUIDialogPVRTimerSettings.cpp

```diff
@@ -561,20 +570,64 @@ void CGUIDialogPVRTimerSettings::Save()
   const CDateTime now(CDateTime::GetCurrentDateTime());
   if (!bStartSet || !bEndSet)
   {
-    if (m_startLocalTime < now)
-      m_startLocalTime = now;
-
-    if (m_endLocalTime < m_startLocalTime)
+    if (m_timerType->SupportsStartEndTime() && //i.e. has start/end time field
+        m_timerType->IsRepeating())            //but has no start day / end day spinners
```

Comment:
Not quite sure whether this condition is correct. What exactly do you want to achieve?